Lib/test/test_inspect.py

```diff
@@ -1507,8 +1507,8 @@ def wrapper(a, b):
 class TestFormatAnnotation(unittest.TestCase):
     def test_typing_replacement(self):
         from test.typinganndata.ann_module9 import ann, ann1
-        self.assertEqual(inspect.formatannotation(ann), 'Union[List[str], int]')
-        self.assertEqual(inspect.formatannotation(ann1), 'Union[List[testModule.typing.A], int]')
+        self.assertEqual(inspect.formatannotation(ann), 'typing.List[str] | int')
+        self.assertEqual(inspect.formatannotation(ann1), 'typing.List[testModule.typing.A] | int')
```

Comment:
This points out we're losing something: When the old `Union` rendered its arguments, it would shorten `typing.List` to `List`, but now it will start using `typing.List`. Of course the best solution is to switch to (builtin) `list`. I guess it changes anyway so we're already accepting that we're breaking users who check the repr of a union type.

Reply:
This is actually because of some weird special casing in `inspect.formatannotation`. I should maybe change it so it also removes "typing." for new-style unions.

Reply:
I just made `typing.Union` the canonical name of the union type, and now the `typing.` gets removed here again.